### ISSUE TITLE
404 response code on step1.html file in keycloak 18.0.0 and 18.0.2

### ISSUE THREAD

**Issue by kashifrana449** [keycloak/keycloak#12766]
### Describe the bug

Keylocak admin console is reloading automatically. I am able to access home page but unable to access administration console. I came to know from network tab that when I click on administration console  /auth/realms/master/protocol/openid-connect/3p-cookies/step1.html is giving 404 in response code.

### Version

18.0.0 and 18.0.2

### Expected behavior

After click on administration console application should move on login page instead reloading the blank page in a loop.

### Actual behavior

A blank page comes and start reloading automatically in a loop when I click on administration console to login.

### How to Reproduce?

- name: KC_HOSTNAME
   value: domain-name
- name: KC_HOSTNAME_STRICT
  value: "false"
- name: KC_HTTP_RELATIVE_PATH
  value: "/auth"
- name: PROXY_ADDRESS_FORWARDING
  value: "true"
- name: KC_HTTP_ENABLED
  value: "true"
- name: KC_PROXY
  value: edge
- name: HOSTNAME-STRICT-HTTPS
  value: "false"

**Comment by sylvaingirardbe** [comment]
I'm experiencing the same after upgrading from 15 to 18.0.2. Keycloak is running on an AKS cluster with more or less the same settings.

**Comment by yasammez** [comment]
I am experiencing the same behaviour in KC 19.0.2. The weird thing is that the browser requests `/auth/realms/master/protocol/openid-connect/3p-cookies/step1.html?version=4ZFSd` which results in a 404. However, `/auth/realms/master/protocol/openid-connect/3p-cookies/step1.html` (without the version query parameter) does exist and can be served …

**Comment by phowells** [comment]
We are also experiencing a siimilar issue since our upgrade from Keycloak 16 to 18.  We are using the Quarkus install.
We updated our React web application to use the 18.0.0 keycloak-js library as part of the update and are now seeing an intermitant 404 when requesting `step1.html`.  We downgraded the keycloak-js library to 10.0.0 (as a troublshooting measure) and we see the intermitatnt 404 when making the `auth` request.  Both requests will work sometimes and both fail sometimes.  Reloading the page after a 404 will sometimes result in a successful reponse but not usually.  We have tested this in both Chrome and Firefox and using incognito mode and the results are the same.

We have set Keycloak to log level `trace` and we do not see any log activity when a 404 is returned.  We do see logs when the request is successful though. 

The 404s are frequent enough to make the application unusable.

**Comment by sschu** [comment]
Have you checked if the error persists in the current version?

**Comment by phowells** [comment]
Upgrading is on our list of things to try but it is a fair bit of work so I
am hoping that someone is familiar with this issue and understands the
cause.  I have not seen anything in the latest releases specific to
addressing this issue.

**Comment by sschu** [comment]
Unfortunately, I am not familiar with it. However, even if this is a bug, it won't be fixed in the old version you are using. In that case you will have to go for the current version anyways. So I would suggest to give this a try as soon as possible, so it can be fixed if the issue persists.

**Comment by phowells** [comment]
Well, we have updated both the  Keycloak server and the keycloak-js library
to 20.0.0 and we are still seeing the 404 response codes intermittently.

**Comment by stianst** [comment]
I've written a util to help debug hostname issues. Download https://github.com/stianst/playground/releases/download/1.0.0/hostname-debug.jar and add to the `providers` directory.

You can then open the tool with `KEYCLOAK_BASE_URL/realms/REALM/hostname-debug` for example `http://localhost:8080/realms/master/hostname-debug`.

If you copy/paste the output here should be easier to help you out. Although I suspect you are facing https://github.com/keycloak/keycloak/issues/15287. You can workaround that by using `hostname-strict-https`, which I see you've tried in your env variables, but you're using `HOSTNAME-STRICT-HTTPS` while it should be `HOSTNAME_STRICT_HTTPS`.

**Comment by githubcek** [comment]
I received same error (on 20.0.1 docker) and if I add KC_PROXY: "edge" than admin console loads :)
OK, it didn't, but when I also added KC_HOSTNAME_ADMIN_URL, it loaded.

I'm using a reverse proxy in front of keycloak.

Just for someone who has keycloak behind a reverse proxy, I have these environment variable set:
KEYCLOAK_ADMIN="admin"
KEYCLOAK_ADMIN_PASSWORD="xxx"
KC_HOSTNAME_STRICT_HTTPS="true"
KC_HOSTNAME="subdomain.domain.com"
KC_PROXY="edge"
KC_HOSTNAME_ADMIN_URL="https://subdomain.domain.com" (same as kc_hostname and at the beggining https://)

and then the admin console loads (otherwise I see a white page and loading spinning).

**Comment by pipinet** [comment]
same error on 20.0.3-0

```yaml
version: '3.8'

services:
  keycloak:
    image: quay.io/keycloak/keycloak:20.0.3-0
    container_name: keycloak
    command: start
    restart: always
    environment:
      KC_HTTP_ENABLED: true
      KC_HTTP_HOST: 0.0.0.0
      KC_HTTPS_CLIENT_AUTH: none
      KC_HOSTNAME_STRICT: false
      KC_HOSTNAME_STRICT_HTTPS: false
      KC_HOSTNAME_URL: http://keycloak.banshan.ren
      KEYCLOAK_ADMIN: admin
      KEYCLOAK_ADMIN_PASSWORD: admin
      KC_DB: postgres
      KC_DB_URL_HOST:  ******
      KC_DB_URL_PORT: 5432
      KC_DB_URL_DATABASE: keycloak
      KC_DB_USERNAME: ******
      KC_DB_PASSWORD: ******
      KC_PROXY: edge
    ports:
      - "18080:8080"
      - "18443:8443"
```

<img width="1746" alt="image" src="https://user-images.githubusercontent.com/6096680/219290254-ed2da555-1281-416f-9022-49c7131242b8.png">

**Comment by AlbinGuo** [comment]
me too.

**Comment by stianst** [comment]
Closing as a duplicate of https://github.com/keycloak/keycloak/issues/14666

**Comment by kasir-barati** [comment]
I do have almost the same issue, But for me it is different a bit. I can see admin console but my react app cannot initialize the keycloak-js. If you are interested read my original question. Note ATM I did not found any answer for it:

Hi there

I am learning keycloak but right now I am receiving this 404 error not found when I try to initialize my keycloak in my react code as you can see here: <https://github.com/kasir-barati/you-say/blob/main/apps/frontend/src/configs/keycloak.config.ts> and the env values are here: <https://github.com/kasir-barati/you-say/blob/main/apps/frontend/.env.example>

You may say to try `http://localhost:8080/auth` instead of `http://localhost:8080` but I did try both and what I get is these error messages in the console:

```cmd
GET http://localhost:8080/realms/you-say-realm/protocol/openid-connect/3p-cookies/step1.html
Object { error: "Timeout when waiting for 3rd party check iframe message." }
```

And When I open <http://localhost:8080/realms/you-say-realm/protocol/openid-connect/3p-cookies/step1.html> in my browser directly I see this page: <https://canada1.discourse-cdn.com/free1/uploads/keycloak/optimized/2X/2/2d86b73b1e3380ff34f825805814ba15af836a22_2_690x392.png>

Any help is really appreciated. I really wanna learn this keycloak but right now here I am blocked. BTW I also configured it with Terraform. Please read README.md if you cloned the project.

**Comment by tspearconquest** [comment]
We upgraded from KC19 Wildfly to KC20, updated our apps to remove references to `/auth` and are running into this issue when trying to login to our apps in a test cluster. The admin console works fine, we can login without issue to the admin console, but cannot get to the login screen within our app.

**Comment by tspearconquest** [comment]
I've added hostname-debug output to #14666 where it will hopefully get a response.

**Comment by michaelnicol** [comment]
It's nearly the second anniversary of this bug, and it still is not fixed.

**Comment by Krst0-it** [comment]
If you haven't solved the issue yet, check the console log in your web browser. I found the following information there: "denied by x-frame-options directive set to deny". I have Keycloak behind nginx acting as a reverse proxy, so I had to add the following header in the nginx configuration:
```
add_header X-Frame-Options "SAMEORIGIN";
```
This might indicate that the problem is not with Keycloak itself, but rather somewhere between your computer and Keycloak, possibly in nginx or something like that.
